services/inputhost/exthost.go

```diff
@@ -625,33 +625,54 @@ func (conn *extHost) aggregateAndSendReplies(numReplicas int) {
 
 				// Note: even if this value is negative, it is ok because we should timeout immediately
 				perMsgTimer.Reset(msgAckTimeout - elapsed)
+
+				// TODO: tighten this further, by ensuring that we received exactly one ack from each replica, i.e. not N from 1 replica
```

Comment:
We cannot receive more than one ACK from a replica because in replicaConnection.go we ensure that is not the case.